### PR TITLE
Fixes various tests of Virtual Machine Scale Set and some for Virtual Machine Unmanaged disks

### PR DESCRIPTION
**Pull request by coderGo93**
- Fixed various tests for Virtual Machine Scale Set
- Removed various tests because of unsupported parameters
- Deprecated some parameters because of unsupported
- Fixes some tests for Virtual Machine Unmanaged disks
- Investigation still ongoing about an error for test `TestAccVirtualMachine_basicLinuxMachineUseExistingOsDiskImage` but the other ones passes.

### CODE REVIEW

File: website/docs/r/virtual_machine_scale_set.html.markdown [hunk 1]
```diff
@@ -129,7 +129,7 @@ The following arguments are supported:
 * `storage_profile_image_reference` - (Optional) A storage profile image reference block as documented below.
 * `extension` - (Optional) Can be specified multiple times to add extension profiles to the scale set. Each `extension` block supports the fields documented below.
 * `plan` - (Optional) A plan block as documented below.
-* `priority` - (Optional) Specifies the priority for the virtual machines in the scale set, defaults to `Regular`. Possible values are `Low` and `Regular`.
+* `priority` - (Optional) **NOT SUPPORTED** Specifies the priority for the virtual machines in the scale set, defaults to `Regular`. Possible values are `Low` and `Regular`.
```
Comment (katbyte) [inline website/docs/r/virtual_machine_scale_set.html.markdown:132]: if this is not supported can we remove it?

File: website/docs/r/virtual_machine_scale_set.html.markdown [hunk 2]
```diff
@@ -140,7 +140,7 @@ The following arguments are supported:
 * `tier` - (Optional) Specifies the tier of virtual machines in a scale set. Possible values, `standard` or `basic`.
 * `capacity` - (Required) Specifies the number of virtual machines in the scale set.
 
-`identity` supports the following:
+`identity`  **NOT SUPPORTED**  supports the following:
```
Comment (katbyte) [inline website/docs/r/virtual_machine_scale_set.html.markdown:143]: if this is not supported can we remove it?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/compute/virtual_machine_scale_set_resource.go [hunk 3]
```diff
@@ -1942,7 +1947,7 @@ func expandazurestackVirtualMachineScaleSetsStorageProfileDataDisk(d *pluginsdk.
 	return &dataDisks
 }
 
-func expandazurestackVirtualMachineScaleSetStorageProfileImageReference(d *pluginsdk.ResourceData) (*compute.ImageReference, error) {
+func expandAzureRmVirtualMachineScaleSetStorageProfileImageReference(d *pluginsdk.ResourceData) (*compute.ImageReference, error) {
```
Comment (katbyte) [inline internal/services/compute/virtual_machine_scale_set_resource.go:1950]: could we change all these AzureRm -> 
```suggestion
func expandVirtualMachineScaleSetStorageProfileImageReference(d *pluginsdk.ResourceData) (*compute.ImageReference, error) {
```

Comment (coderGo93) [reply internal/services/compute/virtual_machine_scale_set_resource.go:1950]: Good eye, will update the naming, thank you

**Review by katbyte — APPROVED** [review]
Thanks @coderGo93 - all VMSS tests pass now!

**Pull request merged**